src/backend/routers/chat.py

```diff
@@ -108,7 +112,7 @@ async def chat(
         should_store,
         managed_tools,
         deployment_config,
-    ) = process_chat(session, chat_request, request)
+    ) = process_chat(session, chat_request, request, agent_id)
```

Comment:
I think for now FE will just have to get the deployment of the agent and pass it into the header.

Starting to realize this def will need to be a new endpoint, but I can follow up on that after.